kernel/src/sched.rs

```diff
@@ -61,16 +61,37 @@ impl Kernel {
     }
 
     /// Something was scheduled for a process, so there is more work to do.
+    ///
+    /// This is only exposed in the core kernel crate.
     pub(crate) fn increment_work(&self) {
         self.work.increment();
     }
 
+    /// Something was scheduled for a process, so there is more work to do.
+    ///
+    /// This is exposed publicly, but restricted with a capability. The intent
+    /// is that external implementations of `ProcessType` need to be able to
+    /// indicate there is more process work to do.
+    pub fn increment_work_public(&self, _capability: &dyn capabilities::MainLoopCapability) {
+        self.work.increment();
+    }
+
     /// Something finished for a process, so we decrement how much work there is
     /// to do.
```

Comment:
```suggestion
    /// Something finished for a process, so we decrement how much work there is
    /// to do.
    ///
    /// This is only exposed in the core kernel crate
```
(to match added annotation for increment_work)